pkg/kubelet/kubelet.go

```diff
@@ -2310,22 +2313,55 @@ func (kl *Kubelet) rejectPod(pod *v1.Pod, reason, message string) {
 // the pod cannot be admitted.
 // allocatedPods should represent the pods that have already been admitted, along with their
 // admitted (allocated) resources.
-func (kl *Kubelet) canAdmitPod(allocatedPods []*v1.Pod, pod *v1.Pod) (bool, string, string) {
+func (kl *Kubelet) canAdmitPod(allocatedPods []*v1.Pod, pod *v1.Pod) (ok bool, reason string, msg string) {
```

Comment:
The original PR also tried to record the metric in `canAdmitPod`, which seems like the obvious choice, but it's a problem because it get's called repeatedly in the place of a resize, which would pollute the metric (https://github.com/kubernetes/kubernetes/pull/125385#discussion_r1631576334)

We could add a dimension to the metric to separate new pods from resizes, but I think it's more useful to just record it in `HandlePodAdditions`, and capture the resize deferred separately.